### PR TITLE
Added symbols loading when loading in-memory modules

### PR DESCRIPTION
**Pull request by thanikTheDev**
This is a modification of PR #108 that only deals with symbol loading when loading in-memory modules.

This might fix related issues
#34
#91 

It at least fixes my issue with Godot symbols not being loaded.

### CODE REVIEW

File: src/metadata/modules.cpp
Comment (viewizard) [inline src/metadata/modules.cpp:492]: https://github.com/dotnet/runtime/blob/d06742c2dbf9ded9c1a896c2a6d0ba1e333a906e/src/coreclr/debug/di/module.cpp#L1227
`GetProcess()` will `AddRef` for `ICorDebugProcess` internal object, should be "released" after end of usage.
Since you have several exit points (`IfFailRet`, `return E_FAIL;` and end of scope), you probably should use `ToRelease<ICorDebugProcess> process;` instead, as we usually do in code for COM objects... or care about COM object release at each exit points.

**Review by viewizard — CHANGES_REQUESTED** [review]

File: src/metadata/modules.cpp
Comment (viewizard) [inline src/metadata/modules.cpp:499]: If I am correct, this mean it copy module from debuggee process into debugger process. Is it copy all modules or in-memory modules only?

Comment (viewizard) [reply src/metadata/modules.cpp:499]: I mean, this looks like you copy all, and this is not really good, since could take a lot of time for big programs on weak devices (like TV or RPI). Plus, if case `ReadMemory` fail with copy for normal module ->`return E_FAIL;`... but why, this is not issue for normal (not in-memory) module that load data from DLL file in this case.

File: src/metadata/modules.cpp
Comment (thanikTheDev) [reply src/metadata/modules.cpp:492]: You're right I completely missed that, I added the tag now.

File: src/metadata/modules.cpp
Comment (thanikTheDev) [reply src/metadata/modules.cpp:499]: I tested it just now and can verify on my end that this section only copies the in memory module. I understand the performance concern but not entirely sure if there is a better way to do it. Least that can be done is limit it to only the in-memory modules.

Comment (viewizard) [reply src/metadata/modules.cpp:499]: As I see in MS docs, 
https://learn.microsoft.com/en-us/dotnet/framework/unmanaged-api/debugging/icordebugmodule-getbaseaddress-method
```
If the module is a native image (that is, if the module was produced by the native image generator, NGen.exe), its base address will be zero.
```
https://learn.microsoft.com/en-us/dotnet/framework/unmanaged-api/debugging/icordebugmodule-getsize-method
```
If the module was produced from the native image generator (NGen.exe), the size of the module will be zero.
```
Did you tested with normal modules, not native images?
As I know, this calls return proper data (not zero) and we also use them in other debugger code (stepper/breakpoint/etc) for Debug built modules (that is not native image for sure).

Not sure how in-memory modules should be detected, but `if (peAddress != 0 && peSize != 0)` not work for this for sure.

Comment (viewizard) [reply src/metadata/modules.cpp:499]: Ohh... miss `IsInMemory` call.

**Comment by viewizard** [comment]
Looks good for me. We move this PR into internal dev repo for more testing with internal CI, this will take some time (have a lot of work now). Will be in next repo sync in upstream.

File: src/metadata/modules.cpp
Comment (viewizard) [inline src/metadata/modules.cpp:491]: BTW, we probably don't need `process` before `if (peAddress != 0 && peSize != 0)` check, since in-memory module could be native image...

Comment (thanikTheDev) [reply src/metadata/modules.cpp:491]: Okay I made that change

**Comment by viewizard** [comment]
I moved this PR into dev repo, all tests passed in CI, waiting for approvals and merge into dev repo now.

**Comment by gbalykov** [comment]
Thanks for your contribution, fix is part of latest release

**Pull request closed without merging**
